src/js/apps/forms/form/form_app.js

```diff
@@ -104,25 +104,37 @@ export default App.extend({
     });
 
     this.listenTo(formService, {
-      'success'(response) {
-        response.set({ _created_at: dayjs().format() });
-        this.responses.unshift(response);
-        this.setState({ responseId: response.id });
+      'submit'() {
+        const saveButtonType = this.getState('saveButtonType');
+
+        if (saveButtonType !== 'saveAndGoBack') return;
 
+        this.loadingModal = Radio.request('modal', 'show:loading');
+      },
+      'success'(response) {
         const saveButtonType = this.getState('saveButtonType');
 
         if (saveButtonType === 'saveAndGoBack') {
-          Radio.request('history', 'go:back');
+          this.listenTo(this.loadingModal, 'destroy', () => {
+            Radio.request('history', 'go:back');
+          });
+
```

Comment:
Moved this listener from the `'submit'` event to the `'success'` event.

If it's placed in the `'submit'` listener, `Radio.request('history', 'go:back');` will be fired when a form error occurs and the modal is destroyed (via `'error'` event). 

We only want the redirect when the  modal is destroyed after a successful form submission. Not when it's destroyed after a form submission error.